src/components/NewGalaxyModal.js

```diff
@@ -0,0 +1,194 @@
+    const { app: oldApp } = this.getOldEnvironmentConfig()
+    return Utils.switchCase(viewMode,
+      ['deleteWarn', () => this.deleteGalaxy()],
+      ['createWarn', () => this.createGalaxy()],
+      [Utils.DEFAULT, () => !!oldApp ? this.setState({ viewMode: 'deleteWarn' }) : this.setState({ viewMode: 'createWarn' })]
+    )
+  }
+
+  render() {
+    const { onDismiss } = this.props
+    const { viewMode } = this.state
+
+    const renderBottomButtons = () => {
+      return h(Fragment, [
+        div({ style: { display: 'flex', margin: '1rem 0 1rem' } }, [
+          div({ style: { flex: 1 } }),
+          this.renderActionButton()
+        ])
+      ])
+    }
+
+    const getMachineDetails = () => {
+      return _.filter(({ name }) => name === 'n1-standard-8', machineTypes)[0]
+    }
+
+    const renderCreateWarning = () => {
+      return h(Fragment, [
+        div({ style: { marginBottom: '1rem' } }, ['Environment consists of an application and cloud compute.']),
+        div({ style: { ...styles.whiteBoxContainer, backgroundColor: colors.accent(.1), boxShadow: Style.standardShadow } }, [
+          div({ style: { display: 'flex', padding: `0.5rem 0 0`, minHeight: 120 } }, [
+            div({ style: { flex: '1', lineHeight: '1.5rem', minWidth: 0, display: 'flex' } }, [
+              span({ style: { marginRight: '0.5rem', marginTop: '0.5rem' } }, [icon('info-circle', { size: 25, color: colors.accent() })]),
+              div([
+                span({ style: styles.headerText }, ['Set up duration']),
+                div({ style: { lineHeight: '24px' } }, [
+                  div(['Creating a cloud environment for Galaxy takes ', span({ style: { fontWeight: 600 } }, ['8-10 minutes.'])]),
+                  div(['You can navigate away, and we will notify you when it\'s ready. '])
+                ]),
+                div({ style: { ...styles.headerText, marginTop: '0.5rem' } }, ['Continuation cost']),
+                div({ style: { lineHeight: '18px' } }, [
+                  div(['Please delete the cloud environment when finished; it will']),
+                  div(['continue to ', span({ style: { fontWeight: 600 } }, ['incur charges ']), 'if it keeps running.'])
+                ])
+              ])
+            ])
+          ])
+        ])
+      ])
+    }
+
+    const renderDeleteWarning = () => {
+      return div({ style: { lineHeight: '22px' } }, [
+        div({ style: { marginTop: '0.5rem' } }, [
+          'Deleting your Cloud Environment will also ',
+          span({ style: { fontWeight: 600 } }, ['delete any files on the associated hard disk']),
+          ' (e.g. results files). Double check that your workflow results were written to ',
+          'the data tab in the workspace before clicking “Delete”'
+        ]),
+        div({ style: { marginTop: '1rem' } }, ['Deleting your Cloud Environment will stop your ',
+          'running Galaxy application and all associated costs. You can create a new Cloud Environment ',
```

Comment:
Based on how Kubernetes works, it isn't necessarily true that stopping your app will stop all costs (that's only the case if you're the last/only app running on the cluster). It will be tricky to communicate the cost model here, but I think we should try.

Reply:
I think it is similar to clusters where the user is not aware of anyone else spending money in the project. For all intents and purposes, if I, as a user, shut down my app, I am no longer spending money in the project. (Similar to clusters, even if there is an underlying expensive K8 cluster that sticks around until the last app is gone.)

We are going ahead and changing the `all associated costs` to `your associated costs` since that is more accurate/correct though.

Reply:
TOL: From my conversation with Brian on slack, it sounds like whoever is paying for the billing project that the galaxy instances are created from will be the one charged. I don't have a lot of experience with the billing side of things, but would it be possible to identify the 'owner' of a billing project and continue to display the cost to them even if their specific instance is deleted? Disclaimer: I don't know how billing projects are handled so that might not make sense. 

Or maybe in the cost breakdown we could just say something (in a tooltip maybe?) indicating that K8 costs are grouped by billing project and won't be fully stopped until the entire K8 cluster is stopped. 

More TOL: Do we know what happens if someone creates a cluster, then someone else in the billing project is using it, but the first person wants to delete it and stop incurring costs?

Reply:
I don't believe anyone can use an app that another person has created. If the app is deleted, it will only delete the K8 cluster if there are no other apps in the project (I believe)

Basically apps are 1-1 with users, and clusters are 1-1 with projects

Reply:
That makes sense. I'm not sure this is a high-value conversation right now, I'm just curious because I don't fully understand the pay structure for the K8 clusters yet. My question is basically: who pays for the K8 cluster if the person who is currently using it is not the same as the person who created it? (E.g. person A creates the first galaxy instance and starts the K8 cluster, person B creates the second galaxy instance, person A deletes their galaxy instance - who pays for the cluster then?) Is it just always the person who owns the billing project? Or maybe there's something I'm missing and that's a moot point…

Reply:
No worries! The answer is that currently I believe everything is always paid for by the billing project/billing project owner, regardless of who is using what.

Reply:
That makes sense, thanks! :)